sos/plugins/openstack_instack.py

```diff
@@ -28,6 +30,16 @@ def setup(self):
         self.add_copy_spec("/home/stack/instackenv.json")
         self.add_copy_spec("/home/stack/undercloud.conf")
 
+        # collect undercloud-debug: undercloud-debug.log will be created so we
+        # need to generate it in a tmp.dir, collect it now and then cleanup it
+        tmpdir = gettempdir()
+        self.get_cmd_output_now("su -s /bin/sh -c 'source ~/stackrc && " +
+                                "/usr/libexec/openstack-tripleo/" +
+                                "undercloud-debug' stack",
+                                timeout=600, runat=tmpdir,
+                                suggest_filename="undercloud-debug.log")
+        remove("%s/undercloud-debug.log" % tmpdir)
```

Comment:
undercloud-debug runs some commands, logs them to the logfile in $PWD and to stdout (via tee). So technically we can collect just the command output and it works.

The problem is the logfile it generates, that is a garbage for us. I fully agree we cant deal with it by writing it to a predictable location (i.e. it contains private keys that sos obfuscate later on).

There are 2 ways how to move forward:

- require the tool to have configurable logfile (i.e. add option --no-logfile for sos purposes)
- call `Plugin.get_cmd_output_path()` with `make=True` and then `call_ext_prog()` that will generate the logfile to proper directory - and ignore stdout